vulnerable_people_form/integrations/ladcode_shielding_advice_lookup.py

```diff
@@ -0,0 +1,57 @@
+import csv
+from datetime import datetime
+import logging
+
+from vulnerable_people_form.form_pages.shared.logger_utils import create_log_message, log_event_names
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_NATIONAL_SHIELDING_ADVICE = 'NO'
+FIELDNAMES = [
+    'ladcode',
+    'ladname',
+    'shielding_advice',
+    'start_datetime_utc'
+]
+
+
+class LocalAuthorityShielding:
+    la_shielding_advice_data = {}
+
+    def __init__(self, shielding_advice_csv_file: str):
+        self.la_shielding_advice_data = {}
+        valid_applicable_records = []
+
+        with open(shielding_advice_csv_file, 'r') as f:
+            la_shielding_advice = csv.DictReader(f, delimiter=',')
+
+            # Find all non-future records
+            for la in la_shielding_advice:
+                now = datetime.utcnow()
+                if datetime.strptime(la['start_datetime_utc'], '%Y-%m-%dT%H:%M:%S') < now:

```

Comment:
Is this right? I don't think this will do the right when the shielding status changes at midnight? I don't think we should be called `__init__` on every request.

Reply:
as Discussed, this is now changed . Essentialy 
- Init only reads and sort the file
- Filtering of latest valid record happens with every call to `get_shielding_advice_by_ladcode`. So when time changes, it will still work without needing to initialize the class and/or session